xgboost_ray/compat/tracker.py

```diff
@@ -0,0 +1,361 @@
+# flake8: noqa
+# Copied from https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/tracker.py
```

Comment:
Do you need to add some licensing notes here?

Reply:
Right, this is Apache 2.0 so probably yeah. I'll quickly check how to attribute this correctly

Reply:
Ok since our own license is also Apache 2.0 we shouldn't run into any problems.

I still added a license header to this file and linked to the correct (hash-identified) file source and license file on the xgboost repo.

Reply:
(pushed but github seems to need some time to update this PR)